src/Workleap.OpenApi.MSBuild/OasdiffManager.cs

```diff
@@ -1,35 +1,71 @@
         this._httpClientWrapper = httpClientWrapper;
         this._processWrapper = processWrapper;
-        this._toolDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
+        this._openApiToolsDirectory = openApiToolsDirectoryPath;
+        this._oasdiffDirectory = Path.Combine(openApiToolsDirectoryPath, "oasdiff", OasdiffVersion);
     }
-
+    
     public async Task InstallOasdiffAsync(CancellationToken cancellationToken)
     {
-        Directory.CreateDirectory(this._toolDirectory);
+        this._loggerWrapper.LogMessage("Starting Oasdiff installation.");
+            
+        Directory.CreateDirectory(this._oasdiffDirectory);
 
         var oasdiffFileName = GetOasdiffFileName();
         var url = $"https://github.com/Tufin/oasdiff/releases/download/v{OasdiffVersion}/{oasdiffFileName}";
 
-        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._toolDirectory, oasdiffFileName), cancellationToken);
+        await this._httpClientWrapper.DownloadFileToDestinationAsync(url, Path.Combine(this._oasdiffDirectory, oasdiffFileName), cancellationToken);
         await this.DecompressDownloadedFileAsync(oasdiffFileName, cancellationToken);
+            
+        this._loggerWrapper.LogMessage("Oasdiff installation completed.");
+    }
+
+    public async Task RunOasdiffAsync(IEnumerable<string> openApiSpecFiles, IEnumerable<string> generatedOpenApiSpecFiles, CancellationToken cancellationToken)
+    {
+        var generatedOpenApiSpecFilesList = generatedOpenApiSpecFiles.ToList();
+        var oasdiffExecutePath = Path.Combine(this._oasdiffDirectory, RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "oasdiff.exe" : "oasdiff");
+
+        foreach (var specFile in openApiSpecFiles)
+        {
+            this._loggerWrapper.LogMessage($"Starting Oasdiff comparison with {specFile}.");
+            
+            var fileName = Path.GetFileName(specFile);
+            var baseSpecRelativePath = Path.GetRelativePath(this._openApiToolsDirectory, specFile);
+            var newSpecRelativePath = Path.GetRelativePath(this._openApiToolsDirectory, generatedOpenApiSpecFilesList.First(x => x.Contains(fileName)));
+            await this._processWrapper.RunProcessAsync(oasdiffExecutePath, new[] { "diff", baseSpecRelativePath, newSpecRelativePath, "--exclude-elements", "description,examples,title,summary", "-f", "text" }, cancellationToken);
+        }
     }
 
     private async Task DecompressDownloadedFileAsync(string oasdiffFileName, CancellationToken cancellationToken)
     {
-        var pathToCompressedFile = Path.Combine(this._toolDirectory, oasdiffFileName);
-        await this._processWrapper.RunProcessAsync("tar", new[] { "-xzf", $"{pathToCompressedFile}", "-C", $"{this._toolDirectory}" }, cancellationToken);
+        var pathToCompressedFile = Path.Combine(this._oasdiffDirectory, oasdiffFileName);
+        if (File.Exists(pathToCompressedFile))
```

Comment:
Should we check if it's the right version? Should we check before calling DownloadFileToDestinationAsync to avoid unecessary duplicate download?

Reply:
There won't be duplicate download because we specifically only download `1.9.2` version. We do a simple check to see if there is an existing file or not before downloading

Reply:
The version is not a parameter? On the CI it won't matter since would start from fresh env but locally if we update the version than the user won't be able to download a new version no?

If we already check when downloading then this condition here should never be true no?